### DISCUSSION TITLE
What does the "docusaurus": "docusaurus" script do?

### DISCUSSION THREAD

**Discussion by ezbeazy** [facebook/docusaurus#8441] [Q&A] [answered]
What is the package.json script "docusaurus": "docusaurus" doing under the hood?

```
package.json

scripts": {
    "docusaurus": "docusaurus",
    "start": "docusaurus start",
    "build": "docusaurus build",
    "swizzle": "docusaurus swizzle",
}
```

**Comment by Josh-Cena** [comment, marked as answer]
It allows you to run `npm run docusaurus docs:version`, because npm doesn't allow you to run binaries from arbitrary packages unless they are declared in the `scripts`. If you are using Yarn, then this should be unnecessary.

**Reply by ezbeazy** [reply]
Is that an npm convention of passing a key name as its own value? or built into docusaurus? and if so where?

What is `docusaurus docs:version` I'm not seeing that in the CLI docs. Thanks!

**Reply by Josh-Cena** [reply]
It's an npm feature. It's the same way you may do `"tsc": "tsc"`. It's simply a way for you to register the command so you can use npm to invoke it.

The reason we need a catch-all `"docusaurus": "docusaurus"` is because plugins can register external commands, which the init template cannot possibly include. The `docs:version` command is from the docs plugin. See: https://docusaurus.io/docs/next/versioning#tagging-a-new-version

**Reply by ezbeazy** [reply]
Essentially that creates a prefix for project specific scripts? Is that in any way aliasing `node someFile.js`?

**Reply by Josh-Cena** [reply]
No. `docusaurus` is a binary provided by the docusaurus package. See:

https://github.com/facebook/docusaurus/blob/1c40d6c821ca937e80626feeb44e9559d3037a47/packages/docusaurus/package.json#L23-L25

**Reply by ezbeazy** [reply]
got it thank you!  In regards to my aliasing question I found my answer in the npm docs https://docs.npmjs.com/cli/v9/using-npm/scripts

"
`[npm run <user defined>]`(https://docs.npmjs.com/cli/v9/commands/npm-run-script)
- pre `<user-defined>`
- `<user-defined>`
- post `<user-defined>`

`[npm start]`(https://docs.npmjs.com/cli/v9/commands/npm-start)
- prestart
- start
- poststart

If there is a server.js file in the root of your package, then npm will default the start command to node server.js. prestart and poststart will still run in this case.
"
